coreapps-web-2.2/src/test/java/org/openmrs/module/coreapps/diagnosis/DiagnosisServiceCompatibilityTest.java

```diff
@@ -0,0 +1,43 @@
+package org.openmrs.module.coreapps.diagnosis;
+
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.openmrs.Diagnosis;
+import org.openmrs.Encounter;
+import org.openmrs.Location;
+import org.openmrs.Patient;
+
+
+public class DiagnosisServiceCompatibilityTest {
+	
+	@Test
+	public void getPrimaryDiagnosis_shouldReturnNotNull() {
+		
+	Encounter encounter = new Encounter();
+    
+	Date date = new Date();
+	Location location = new Location(1);
+	Patient patient = new Patient(1);
+	
+	Diagnosis diagnosis = new Diagnosis();
+	diagnosis.setDiagnosisId(2);
+	
+	Set<Diagnosis> diag = new HashSet<Diagnosis>();
+	diag.add(diagnosis);
+	
+	encounter.setEncounterDatetime(date);
+	encounter.setLocation(location);
+	encounter.setPatient(patient);
+	DiagnosisServiceCompatibility2_2 diagnosisServiceCompatibility = new DiagnosisServiceCompatibility2_2();
+
+	Assert.assertEquals(1,diagnosisServiceCompatibility.getPrimaryDiagnoses(encounter).size());	
```

Comment:
hi @dkayiwa  @ibacher ,i get this test breaking with these logs https://pastebin.com/jaJU0hBr ie Service not found: interface org.openmrs.api.DiagnosisService, any clue on fixing this

Reply:
@HerbertYiga ,usually that a hard error to debug , but its more like at the test Run time , its a different OpenMRS version that runs ,rather than 2.2 . and remember the [Diagnosis service](https://github.com/openmrs/openmrs-core/blob/2.2.0/api/src/main/java/org/openmrs/api/context/ServiceContext.java#L228-L236) is only loaded from OpenMRS 2.2 
it seems it runs the OpenMRS version in the Parent POM  when runnign you tests.

I havent had more time to look at your POM but play more around with your POM file` coreapps-web-2.2/pom.xml`  .

A Quick example , After you import your project in Eclipse , look at the Library jars that have been loaded, you may find mixed OpenMRS versions  in your submodule 2.2 ,and its supposed to be one Version.
Look at the anwers alongside this Talk Post https://talk.openmrs.org/t/failing-to-run-tests-because-of-conflicting-bean-definition-exception/17412

Reply:
thanks @mozzy11 have really found the comments help full

Reply:
@mozzy11 hei i got unblocked on this, with your comments i got realize that i hadn't created my submodule well ,my pom had conflicting files thanks